src/hotspot/share/cds/classListParser.cpp

```diff
@@ -85,13 +85,21 @@ bool ClassListParser::is_parsing_thread() {
 }
 
 ClassListParser::~ClassListParser() {
-  if (_file) {
+  if (_file != NULL) {
     fclose(_file);
   }
   Atomic::store(&_parsing_thread, (Thread*)NULL);
+  delete _indy_items;
+  delete _interfaces;
+  cleanup_table();
```

Comment:
I just realized that the `_id2klass_table` contains only `InstanceKlass*` values, so there's no need to clean up the values themselves (the InstanceKlass should keep on living after the table is deallocated).

It's the entries of `_id2klass_table` that need to be freed, but that should be done separately. See [JDK-8267752](https://bugs.openjdk.java.net/browse/JDK-8267752)

So I think the changes related to `_id2klass_table` in this PR are no longer needed. Sorry for pointing you to the wrong direction. The rest looks OK to me.

Reply:
You still have to clean up the id2hashtable entries that are CHEAP allocated.   But I've got that in my upcoming PR.

Reply:
Ioi, Coleen, Yumin, thanks for the review.
I've reverted the changes related to `_id2klass_table`.